R/format_PEW_temp.R

```diff
@@ -0,0 +1,541 @@
+    # #### BUT REMOVE IT BEFORE FINAL OUPUT!!
+
+  return(Capture_data)
+
+}
+
+create_individual_PEW <- function(data){
+
+  Individual_data <-
+    data %>%
+    #### NOTE: Keep rows without Capture date
+    #### Format and create new data columns
+    dplyr::mutate(RingAge = ifelse(Age_observed == 1, "chick", "adult"),
+                  #### We actually do not have LaidDate, for several nests
+                  # there is only DateEgg1.
+                  # In metadata, there is a note that second and other broods are not recorded,
+                  # so can we assume that each brood is the first and only one?
+                  # BroodIDLaid = ifelse(RingAge == "chick",
+                  #                      paste(BreedingSeason, LocationID,
+                  #                            paste0(lubridate::year(LaidDate),
+                  #                                   sprintf("%02d", lubridate::month(LaidDate)),
+                  #                                   sprintf("%02d", lubridate::day(LaidDate))),
+                  #                            sep = "_"),
+                  #                      NA_character_)
+                  BroodIDLaid    = ifelse(RingAge == "chick",
+                                          paste(BreedingSeason, LocationID, sep = "_"),
+                                          NA_character_),
+                  BroodIDFledged = BroodIDLaid) %>%
+    dplyr::group_by(IndvID) %>%
+    dplyr::mutate(Sex_calculated = purrr::map_chr(.x = list(unique(na.omit(Sex_observed))), .f = ~{
+                                                            if(length(..1) == 0){
+                                                              return(NA_character_)
+                                                            } else if(length(..1) == 1){
+                                                              return(..1)
+                                                            } else {
+                                                              return("C")
+                                                            }
+                                                          }),
+                  Sex_genetic = NA_character_) %>%
+    dplyr::arrange(CaptureDate) %>%
+    dplyr::summarize(Species = first(Species),
+                     PopID = "PEW",
+                     IndvID = first(IndvID),
+                     RingSeason = first(BreedingSeason),
+                     RingAge = first(RingAge),
+                     BroodIDLaid = first(BroodIDLaid),
+                     BroodIDFledged = first(BroodIDFledged),
+                     Sex_calculated = first(Sex_calculated),
+                     Sex_genetic = first(Sex_genetic)) %>%
+    dplyr::ungroup() %>%
+    #### Final arrangement
+    dplyr::select(IndvID, Species, PopID, BroodIDLaid, BroodIDFledged,
+                  RingSeason, RingAge, Sex_calculated, Sex_genetic)
```

Comment:
`Sex_calculated` is just intended to return a single value per individual, so `summarise` would be more appropriate. Your current code doesn't work for some individuals. For example, 13617071 was caught as a chick in nest 2016_91, but it is not given this as BroodIDLaid because the CaptureDate of this record is NA (but see my comment above. The below code should be more robust.

 ```suggestion
Individual_data <- data %>% 
    group_by(IndvID) %>% 
    dplyr::summarise(Sex_calculated = purrr::map_chr(.x = list(unique(na.omit(Sex_observed))), .f = ~{
        if(length(..1) == 0){
            return(NA_character_)
        } else if(length(..1) == 1){
            return(..1)
        } else {
            return("C")
        }
    }),
    Sex_genetic = NA_character_,
    Species = first(Species),
    PopID = "PEW",
    #IndvID = first(IndvID), #Not needed because it's the grouping factor
    RingSeason = min(BreedingSeason),
    RingAge = ifelse(min(Age_observed) == 1, "chick", "adult"),
    BroodIDLaid = ifelse(RingAge == "chick",
                         paste(BreedingSeason, LocationID, sep = "_"),
                         NA_character_),
    BroodIDFledged = BroodIDLaid) %>%
    dplyr::ungroup() %>% 
    dplyr::select(IndvID, Species, PopID, BroodIDLaid, BroodIDFledged,
                  RingSeason, RingAge, Sex_calculated, Sex_genetic)
```